osmoutils/cosmwasm/helpers.go

```diff
@@ -61,12 +61,24 @@ func Query[T any, K any](ctx sdk.Context, wasmKeeper WasmKeeper, contractAddress
 	// Check remaining gas in parent context and use the lesser of the query gas limit and remaining gas
 	gasLimit := min(ctx.GasMeter().GasRemaining(), wasmKeeper.QueryGasLimit())
 	childCtx := ctx.WithGasMeter(storetypes.NewGasMeter(gasLimit))
-	responseBz, err := wasmKeeper.QuerySmart(childCtx, sdk.MustAccAddressFromBech32(contractAddress), bz)
-	if err != nil {
-		return response, err
-	}
 
-	ctx.GasMeter().ConsumeGas(childCtx.GasMeter().GasConsumed(), "query smart")
+	// Execute the query with proper gas tracking and panic recovery
+	var queryErr error
+	var responseBz []byte
+	func() {
+		defer func() {
```

Comment:
why is the double func needed? If the annonymous function gets called immediately, isn't this the same as just declaring the defer directly?

Reply:
@nicolaslara  Will do!